cantools/database/can/message.py

```diff
@@ -1085,19 +1064,33 @@ def decode(self,
 
         """
 
-        if self.is_container:
-            if decode_containers:
-                return self._decode_contained(data,
-                                              decode_choices,
-                                              scaling,
-                                              allow_truncated)
-            else:
-                raise DecodeError(f'Message "{self.name}" is a container '
-                                  f'message, but decoding such messages has '
-                                  f'not been enabled by setting the '
-                                  f'decode_containers parameter to True')
+        if self.is_container and decode_containers:
```

Comment:
Evaluate the computationally inexpensive boolean first.

```suggestion
        if decode_containers and self.is_container:
```